src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -1500,6 +1504,28 @@ private ulong GetGraphNodeKey(int classIndex, int procIndex)
             return (((ulong)ci) << 32) | pi;
         }
 
+        internal void Purge(List<AST.AssociativeAST.AssociativeNode> oldNodes)
```

Comment:
Maybe add some comments here? In general, the team try to have all the new functions covered by comments